### DISCUSSION TITLE
Adding support for new Sonoff RF Bridge V2

### DISCUSSION THREAD

**Discussion by laurentppol** [arendst/Tasmota#15409] [Q&A] [answered]
Hi there, 
I got an RF Bridge V2. Using Tasmota 10.0.0 and setting module type to Bridge gives no output on console.
I just hooked up 2 USB->serial dongles on ESP's Rx and Tx lines, found (suppose, not 100% sure) some 19200 8N1 conversations between ESP and companion MCU.
Looks like simple protocol. 
I can discern RF codes I learned using Bridge V1.
But it is not working (now) with Tasmota code ("as is"), so they must have changed something.
I do NOT want to use "hardware hack" (connecting RF Rx and Tx to ESP pins), it worked for V1, but I want no HW mod (except headers) to this module.

So please help, 
starting by telling which piece of code (*.ino I guess, C/C++) is in use when setting module to RF Bridge?
Will look at this to find if I can change to adapt to new Itead protocol.

**Comment by ascillato** [comment]
Hi,

Sonoff RfBridges (R1 and R2) do work. If you have the R2.2 it works too (you can see the RF codes coming in the Tasmota console) but Portish Firmware can't be flashed there.

If you see nothing on the console check if the Tasmota module configuration is correct and if the internal switch is at ON.

The request for supporting the new R2.2 that has a different RF chip is at: https://github.com/arendst/Tasmota/discussions/13283

And a discussion about the R2.2 is at: https://github.com/arendst/Tasmota/discussions/14783

**Comment by laurentppol** [comment]
I heva seen both threads You mention.
They seem a bit stalled.
I have R2 v2.2, (white one), no switch and design in #13283 , RF & WiFi antennas on PCB.

As You may read I want to do some developpment, but don't want to reinvent the wheel :)
And simply don't know where (which code module) to start.

IF (as I can see in ESP Rx) codes transmitted from companion MCU are similar (at least 3 bytes from "old" RFRx and/or MCU were encoded into MQTT messages / console), so there may be only need to slightly adapt ESP code to understand "new" ITEAD protocol.

W/o hardware mods and / or reflashing companion MCU.

**Comment by ascillato** [comment]
> I want to do some developpment, but don't want to reinvent the wheel

Cool, so maybe it is better to continue in the original Idea discussion for this device at https://github.com/arendst/Tasmota/discussions/13283 where you can receive help from the previous persons that have posted some info before. So as to concentrate all the effort in just one place instead of being scattered in several similar opened discussions.

**Comment by laurentppol** [comment, marked as answer]
Ok, will copy / paste my 2 posts in the thread https://github.com/arendst/Tasmota/discussions/13283.
